sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/InMemoryColumnarQuerySuite.scala

```diff
@@ -429,4 +429,19 @@ class InMemoryColumnarQuerySuite extends QueryTest with SharedSQLContext {
       checkAnswer(agg_without_cache, agg_with_cache)
     }
   }
+
+  test("SPARK-22249: IN should work also with cached DataFrame") {
```

Comment:
This test case is just an end-to-end test. This test will still pass if the optimizer has a change. We also need a unit test case.

Reply:
do you mean this test case is not enough and I should add a test case to check the proper behavior of the optimizer after the change?